### ISSUE TITLE
Test Case 10 Infinite Loop

### ISSUE THREAD

**Issue by kyledilao777** [nus-cs2030/2223-s2#386]
# Summary

Hi I'm having issues with Test Case 10 which somehow has an infinite loop... I tried looking at past issues also but i still can't solve mine. Anyone can help me pls :D?

## Description

Something to do with the processing of multiple wait events and transition to serve event. I tried to change the capacity of queue(qcap) condition or the server side also but still cant resolve the issue

## Code Snippets
```
public Pair<Event, ImList<Server>> getNextState(ImList<Server> sl, Supplier<Double> st) {
        //update server
        Server updated = sl.get(s.getID() - 1);
        
        //serve event if not next in line
        if (updated.getqcap() == 1) {
            Serve serve = new Serve(super.getID(), updated.getEndTime(), s, st);
            updated = updated.serve(super.getStartTime(), updated.getEndTime(), serve.getStartTime() - arrivalTime);
            return new Pair<Event, ImList<Server>>(serve, sl.set(s.getID() - 1, updated));

        } else {
            double startTime = super.getStartTime();
            Wait wait = new Wait(super.getID(), updated.getEndTime(), startTime, updated, true);
            updated = updated.update(startTime);
            return new Pair<Event, ImList<Server>>(wait, sl.set(s.getID() - 1, updated));
        }
    }
```
## Screenshots

Screenshots (if any)

**Comment by kenzverylayz** [comment]
hi it could be that the queue for the server not being updated hence the customer is never actually next in line

**Comment by easypzlemonsqueezy** [comment]
not sure if you have an intermediate wait event but the criteria to transition from intermediatewait to serve might be too specific and your customer is forever stuck in the intermediatewait event, hope this helps!

**Comment by nigel-tannn** [comment]
Good idea to include an intermediate wait event that signifies a shift in the customer queue, i.e. that the customers in the queue are moving forward, every time a customer is done being served.  This intermediate wait event can keep adding new intermediate wait events to the PQ until it the customer is at the front of the queue.

**Comment by mavisczx** [comment]
Hello I am also facing this issue, have you solved yours?

**Comment by Danpng01** [comment]
Hi Kyle, one possible error is that your conditions to catch the event to return a serve has a gap that causes this customer to pass through! Apart from checking the server timings, you can check for minor details such as ( < vs <= ) :)
